app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -1,27 +1,138 @@
 package com.github.featuredetectandroid
 
+import android.Manifest
+import android.content.pm.PackageManager
 import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.activity.viewModels
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageAnalysis.OUTPUT_IMAGE_FORMAT_YUV_420_888
+import androidx.camera.lifecycle.ProcessCameraProvider
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material.FloatingActionButton
-import androidx.compose.ui.Alignment
+import androidx.compose.material.Text
 import androidx.compose.ui.Modifier
+import androidx.core.content.ContextCompat
+import com.github.featuredetectandroid.ui.GrayscaleView
+import com.github.featuredetectandroid.ui.GrayscaleViewModel
 import com.github.featuredetectandroid.ui.theme.Theme
+import com.github.featuredetectandroid.utils.PhotoAnalyzer
+import java.util.concurrent.Executors
+
+private const val TAG = "MainActivity"
 
 class MainActivity : ComponentActivity() {
+    private val cameraExecutor = Executors.newSingleThreadExecutor()
+    private val imageViewModel by viewModels<GrayscaleViewModel>()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        tryStartCamera()
         setContent {
             Theme {
                 Box(Modifier.fillMaxSize()) {
-                    FloatingActionButton(
-                        onClick = { /*TODO*/ },
-                        modifier = Modifier.align(Alignment.BottomEnd)
-                    ) { }
+                    if (ContextCompat.checkSelfPermission(
+                            this@MainActivity,
+                            Manifest.permission.CAMERA
+                        ) == PackageManager.PERMISSION_GRANTED
```

Comment:
It seems like this check is run only when the app start and isn't rerun after the permission is granted. So, the following happens:
- I start the app the first time
- It asks me for permissions and this check fails
- I grant the permissions
- Camera view doesn't appear because the check is not rerun

And I need to restart the app to see the output :)

Reply:
Fixed, if I understood you correctrly.

Reply:
Nope, this still happens. Do the following:
1. Remove the app's camera permission (or just reinstall it).
2. Start the app.
3. Grant camera permissions.
4. Still see the "Camera permission required" text.

I think, you need to create a `Boolean` field in your `ViewModel` and track permission status there sp that Compose sees its updates.

Reply:
Fixed. Also added switching between windos support.